common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -145,10 +145,24 @@
 
     private static final String ANDROID_KEY_STORE = "AndroidKeyStore";
 
-    private static final Set<String> BROKER_PACKAGE_NAMES = new HashSet<String>() {{
-        add(AZURE_AUTHENTICATOR_APP_PACKAGE_NAME);
-        add(COMPANY_PORTAL_APP_PACKAGE_NAME);
-    }};
+    /**
+     * Type of Secret key to be used.
+     */
+    public enum KeyType {
+        LEGACY_AUTHENTICATOR_APP_KEY,
+        LEGACY_COMPANY_PORTAL_KEY,
+        ADAL_USER_DEFINED_KEY,
+        KEYSTORE_ENCRYPTED_KEY
```

Comment:
We are planning to store the new key to keystore but just in case it changes we should name something LATEST_LOCALGEN_KEY

Reply:
If we're going to change, it's likely that we'll have to introduce a new one (i.e. as we have to 'decrypt' with the 'old type' and 'encrypt' them with the new type.

Reply:
If we're just going to change how we store the very key, in that case I'd still prefer change the name at that point. I personally found LATEST_LOCALGEN_KEY too generic (i.e. if we introduce hardware-backed encryption, then it's still technically 'locally generated').